### ISSUE TITLE
Google Speech Commands and BindsNET

### ISSUE THREAD

**Issue by hansemandse** [BindsNET/bindsnet#467]
Hi! I am trying to train a very small feed forward network much alike your example with MNIST with the dataset replaced by Google Speech Commands (GSCD), but I can't seem to make the network learn anything (see `train.py`). My source code is available [here](https://github.com/hansemandse/KWSonSNN/tree/master/netwkdev). 

The network I am using is a scaled version of the `DiehlAndCook2015` network (see `kwsonsnn/model.py`) made to fit on a custom hardware accelerator. It works with MNIST achieving roughly 70% accuracy in a single epoch. I believe my dataset (see `kwsonsnn/dataset.py`) is correctly preprocessed in a style similar to your implementation of Spoken MNIST. The regular PyTorch model included in the repo (see `trainpt.py`) achieves 90% accuracy on GSCD indicating that data is not to blame.

I have tried multiple different parameter configurations in terms of weight normalization and learning rates, but they do not seem to improve the network's performance.

So, am I doing something wrong? And do you by any chance have some benchmark results for either Spoken MNIST or GSCD produced with BindsNET?

System specifications:
- Python3 3.7.6
- BindsNET 0.2.7
- numpy 1.18.1
- PyTorch 1.6.0+cpu

**Comment by Hananel-Hazan** [comment]
Thank you for using BindsNET, I think this is more of a discussion and not an issue in BindsNET. 

> am I doing something wrong?

Its hard to say, tuning a network parameters (any network SNN or ANN) is not an easy task and require a lot of trial and error from your part. I can only try to help. I will start by upgrading BindsNET to 0.2.8 (instruction in the main repo page).
Then, please add some plots: weights, voltage curves of the neurons, spikes of #neuron vs time, and input reconstruction. Hopefully the plots will show you how to approach the parameter tuning task.

A brief glimpse on your code, I think that your normalization value is too high. Note that the normalization value choosen by `DiehlAndCook2015` was 10% of the input neurons, that also reflect the B/W pixel ratio in MNIST.

 I couldn't spot an error 


>  do you by any chance have some benchmark results for either Spoken MNIST or GSCD produced with BindsNET?

Sadly, no, but If you run them, I will be happy to add them to the repo with your name as a contributor.

**Comment by hansemandse** [comment]
Thanks for replying so quickly!

I have added the plot code from the MNIST example to my example, but the `plot_inputs` function fails to update intermediate layer spikes when called with non-`None` image/axes parameters. The input spikes plot works, but the plots for intermediate layers are left empty (even with clamped outputs). Any idea on how to fix that?

I'll look into making a PR with GSCD once I have something working :)

**Comment by Hananel-Hazan** [comment]
That is strange, did you check it manually? did you inspect the values of the neuron voltage? when calming, the voltage in the neuron change, that need to be show in the plots 
I believe you can try to see how it works with the other examples.

**Comment by hansemandse** [comment]
Cloning the BindsNET repo and running the supervised MNIST example does the same thing. The spikes for the input layer `X` are updated correctly, but the other layers show spikes only for the first example.

**Comment by Hananel-Hazan** [comment]
I am unable to replicate this. 
Running the supervise MNIST example shows 

![Untitled](https://user-images.githubusercontent.com/3954715/113946299-948e4480-97f7-11eb-9951-3290a4b6fb95.png)

@petermarathas can you take a look on this please

**Comment by petermarathas** [comment]
I see something similar when I run the supervised MNIST example. For me the Ai/Ae spikes will only appear for higher classes (8 and 9 so far).

**Comment by Hananel-Hazan** [comment]
@hansemandse can you check to see if its solve the issue?

**Comment by Hananel-Hazan** [comment]
I am closing this for now, seems that the issue as been resolved
